risc0/core/elf.cpp

```diff
@@ -57,15 +59,36 @@ struct ProgHeader {
 
 } // namespace
 
-uint32_t loadElf(const std::string& name, uint32_t maxMem, std::map<uint32_t, uint32_t>& memOut) {
+std::vector<uint8_t> loadFile(const std::string& path) {
+  std::ifstream is(path, std::ios::binary);
+  auto check_err = [&](const char* msg) {
+    if (is.fail() || is.bad()) {
+      std::stringstream ss;
+      ss << "Could not " << msg << " ELF: " << path;
+      throw std::runtime_error(ss.str());
+    }
+  };
```

Comment:
Would it be sufficient to use `is.exceptions(ios_base::failbit | ios_base::badbit)`?

https://en.cppreference.com/w/cpp/io/basic_ios/exceptions